modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/ShareFragment.kt

```diff
@@ -4,124 +4,264 @@ import android.os.Bundle
-        binding.buttonOpenFileIn.setOnClickListener {
-            if (podcast != null && episode is PodcastEpisode) {
-                SharePodcastHelper(
-                    podcast,
-                    episode,
-                    episode.playedUpTo,
-                    requireContext(),
-                    ShareType.EPISODE_FILE,
-                    SourceView.PLAYER,
-                    analyticsTracker
-                ).sendFile()
+            Divider(
+                color = Color.LightGray.copy(0.3f),
+                thickness = 1.dp,
+            )
+            Divider(thickness = 1.dp)
+            if (podcast != null && episode is BaseEpisode) {
+                ShareButton(
+                    label = IR.string.podcast_share_episode,
+                    onClick = onShareEpisode
+                )
+                Divider(
+                    color = Color.LightGray.copy(0.3f),
+                    thickness = 1.dp,
+                )
+                ShareButton(
+                    label = IR.string.podcast_share_current_position,
+                    onClick = onShareCurrentPosition
+                )
+            }
+            if (podcast != null && episode is BaseEpisode && episode.isDownloaded) {
+                ShareButton(label = IR.string.podcast_share_open_file_in, onClick = onShareOpenFileIn)
             }
-            close()
         }
-
-        binding.sharePodcast.isVisible = podcast != null
-        binding.shareEpisode.isVisible = episode != null
-        binding.shareCurrentPosition.isVisible = episode != null
-        binding.openFileIn.isVisible = episode != null && episode.isDownloaded
-
-        return binding.root
     }
+}
 
-    @Suppress("DEPRECATION")
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
+@Composable
+private fun ShareButton(
+    label: Int,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Box(
+        modifier = modifier
+            .fillMaxWidth()
+            .clickable(
+                interactionSource = remember { MutableInteractionSource() },
+                indication = rememberRipple(color = Color.LightGray)
+            ) { onClick() }
+            .height(64.dp)
+            .padding(start = 32.dp)
+            .wrapContentSize(Alignment.CenterStart)
+    ) {
+        Text(
+            text = stringResource(id = label),
+            color = MaterialTheme.theme.colors.playerContrast01,
+            fontFamily = FontFamily.SansSerif,
+            fontSize = 16.sp
+        )
+    }
+}
 
-        viewModel.playingEpisodeLive.observe(viewLifecycleOwner) { (_, backgroundColor) ->
-            applyColor(theme, backgroundColor)
-        }
+private val PillSize = DpSize(width = 48.dp, height = 4.dp)
+private val PillCornerRadius = 10.dp
 
-        (dialog as? BottomSheetDialog)?.behavior?.state = BottomSheetBehavior.STATE_EXPANDED
-    }
+@Composable
+private fun Pill(
```

Comment:
There can be reusable components that can be extracted out like this `Pill` and the component for each option.  Can you describe the challenges you had in using the `OptionsDialog()` that you initially proposed to use?

This will help us decide the architecture for the rest of the "Playback Effects" dialogs.

Reply:
I agree, the pill can be reused. Regarding the Options Dialog, it would work but one concern: 
If we migrate all fragments using `BaseDialogFragment()` to `optionsDialog()`, we would have to add new rows for each specific fragment. That means for each fragment that is being migrated, we would add new custom composables on the dialog. But if this would not be a problem, it would work.

Reply:
> That means for each fragment that is being migrated, we would add new custom composables on the dialog. 

Could we pass a custom option component (which is only shown on a single screen) to the `OptionsDialog()` so that we don't increase the class size too much and yet be able to have a common class across all these option fragments?

@mchowning, @geekygecko any thoughts?

Reply:
@Mzazi25 👋 

I had a brief discussion with the team, with `OptionsDialog()` we'll be able to reuse common components and see some advantages. Can we give it a try?

Reply:
Sure thing. Let me work on that